lib/internal/callbackify.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+module.exports = callbackify;
+
+const promises = require('internal/promises');
+const BuiltinPromise = promises.Promise;
+
+function callbackify(fn) {
+  Object.defineProperty(inner, 'name', {
+    writable: false,
+    enumerable: false,
+    configurable: true,
+    value: fn.name + 'Callback'
+  });
+  return inner;
+
+  function inner() {
+    const cb = arguments[arguments.length - 1];
+    if (typeof cb !== 'function') {
+      cb = (err) => { throw err; };
+    }
+    try {
+      BuiltinPromise.resolve(fn.apply(this, arguments)).then(
+        (result) => cb(null, result),
```

Comment:
Or, you know, `new BuiltinPromise(resolve => resolve(fn.apply(this, arguments))` - whichever looks better.